### PR TITLE
Update documentation for the new version

### PR DESCRIPTION
**Pull request by SBoudrias**
Feel free to commit directly to this branch for typos and such. I'll squash everything before merging.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,26 +1,102 @@
 # Node Generator [![Build Status](https://secure.travis-ci.org/yeoman/generator-node.svg?branch=master)](https://travis-ci.org/yeoman/generator-node)
 
-> Create a Node.js module
+`generator-node` creates a base template to start a new Node.js modules.
```
Comment (arthurvr) [inline README.md]: `module`, not `modules`

File: README.md [hunk 2]
```diff
@@ -1,26 +1,102 @@
...
 ```
 
 *Note that this template will generate files in the current directory, so be sure to change to a new directory first if you don't want to overwrite existing files.*
 
+That'll generate a project with all the common tools setup. This includes:
+
+- Filled `package.json` file
+- [gulp](http://gulpjs.com/) test runner
+- [mocha](http://mochajs.org/) unit test
+- [jshint](http://jshint.com/) linting
```
Comment (arthurvr) [inline README.md]: JSHint

File: README.md [hunk 3]
```diff
@@ -1,26 +1,102 @@
...
+- [jscs](http://jscs.info/) code style checking
```
Comment (arthurvr) [inline README.md]: didn't they write it as `JSCS`?

File: README.md [hunk 4]
```diff
@@ -1,26 +1,102 @@
...
+- [Istanbul](https://gotwarlost.github.io/istanbul/) code coverage (optionnally tracked on [Coveralls](https://coveralls.io/))
+- [TravisCI](https://travis-ci.org/) continuous integration (optionnal)
```
Comment (arthurvr) [inline README.md]: `Travis CI`

File: README.md [hunk 5]
```diff
@@ -1,26 +1,102 @@
...
+You can also directly use mocha to run test on single files:
+
+```
+$ npm -g install mocha
+$ mocha test/name.js
+```
+
+### Publishing your code
+
+Once you're tests are passing (ideally with a TravisCI green run), you might be ready to publish your code to npm. We recommend you using [npm version]() to tag release correctly.
```
Comment (arthurvr) [inline README.md]: `Travis CI`

Comment (arthurvr) [inline README.md]: >[npm version\]()

Did you want to link to the api docs? Or make this a `<code>` snippet?

**Comment by arthurvr** [comment]
I just pushed some really small fixes.

**Comment by hemanth** [comment]
:+1:

**Comment by 9965buzzer** [comment]
:+1:

**Pull request merged**